commit c481774298246c2bc2d2c66fa736ce2d4cae69d1
Author: Hauke Mehrtens <hauke@hauke-m.de>
Date:   Mon Mar 13 20:07:14 2017 +0100

    curl: update to version 7.53.1
    
    This fixes the following security problem:
    * CVE-2017-2629 SSL_VERIFYSTATUS ignored
    
    Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

diff --git a/lede/package/network/utils/curl/Makefile b/lede/package/network/utils/curl/Makefile
index 950044a226..d9e906c282 100644
--- a/lede/package/network/utils/curl/Makefile
+++ b/lede/package/network/utils/curl/Makefile
@@ -8,8 +8,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=curl
-PKG_VERSION:=7.52.1
-PKG_RELEASE:=6
+PKG_VERSION:=7.53.1
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=http://curl.haxx.se/download/ \
@@ -18,7 +18,7 @@ PKG_SOURCE_URL:=http://curl.haxx.se/download/ \
 	ftp://ftp.planetmirror.com/pub/curl/ \
 	http://www.mirrormonster.com/curl/download/ \
 	http://curl.mirrors.cyberservers.net/download/
-PKG_HASH:=d16185a767cb2c1ba3d5b9096ec54e5ec198b213f45864a38b3bda4bbf87389b
+PKG_HASH:=1c7207c06d75e9136a944a2e0528337ce76f15b9ec9ae4bb30d703b59bf530e8
 
 PKG_LICENSE:=MIT
 PKG_LICENSE_FILES:=COPYING
diff --git a/lede/package/network/utils/curl/patches/001-curl-https-openssl-fix.patch b/package/network/utils/curl/patches/001-curl-https-openssl-fix.patch
deleted file mode 100644
index 259f618d46..0000000000
--- a/lede/package/network/utils/curl/patches/001-curl-https-openssl-fix.patch
+++ /dev/null
@@ -1,34 +0,0 @@
-From a7b38c9dc98481e4a5fc37e51a8690337c674dfb Mon Sep 17 00:00:00 2001
-From: Daniel Stenberg <daniel@haxx.se>
-Date: Mon, 26 Dec 2016 00:06:33 +0100
-Subject: [PATCH] vtls: s/SSLEAY/OPENSSL
-
-Fixed an old leftover use of the USE_SSLEAY define which would make a
-socket get removed from the applications sockets to monitor when the
-multi_socket API was used, leading to timeouts.
-
-Bug: #1174
----
- lib/vtls/vtls.c | 4 ++--
- 1 file changed, 2 insertions(+), 2 deletions(-)
-
---- a/lib/vtls/vtls.c
-+++ b/lib/vtls/vtls.c
-@@ -484,7 +484,7 @@ void Curl_ssl_close_all(struct Curl_easy
-   curlssl_close_all(data);
- }
- 
--#if defined(USE_SSLEAY) || defined(USE_GNUTLS) || defined(USE_SCHANNEL) || \
-+#if defined(USE_OPENSSL) || defined(USE_GNUTLS) || defined(USE_SCHANNEL) || \
-     defined(USE_DARWINSSL) || defined(USE_NSS)
- /* This function is for OpenSSL, GnuTLS, darwinssl, and schannel only. */
- int Curl_ssl_getsock(struct connectdata *conn, curl_socket_t *socks,
-@@ -518,7 +518,7 @@ int Curl_ssl_getsock(struct connectdata
-   (void)numsocks;
-   return GETSOCK_BLANK;
- }
--/* USE_SSLEAY || USE_GNUTLS || USE_SCHANNEL || USE_DARWINSSL || USE_NSS */
-+/* USE_OPENSSL || USE_GNUTLS || USE_SCHANNEL || USE_DARWINSSL || USE_NSS */
- #endif
- 
- void Curl_ssl_close(struct connectdata *conn, int sockindex)
diff --git a/lede/package/network/utils/curl/patches/200-no_docs_tests.patch b/lede/package/network/utils/curl/patches/200-no_docs_tests.patch
index 139c05e64b..2fbcf59a41 100644
--- a/lede/package/network/utils/curl/patches/200-no_docs_tests.patch
+++ b/lede/package/network/utils/curl/patches/200-no_docs_tests.patch
@@ -1,6 +1,6 @@
 --- a/Makefile.am
 +++ b/Makefile.am
-@@ -153,7 +153,7 @@ CLEANFILES = $(VC6_LIBDSP) $(VC6_SRCDSP)
+@@ -152,7 +152,7 @@ CLEANFILES = $(VC6_LIBDSP) $(VC6_SRCDSP)
  bin_SCRIPTS = curl-config
  
  SUBDIRS = lib src include
@@ -11,7 +11,7 @@
  pkgconfig_DATA = libcurl.pc
 --- a/Makefile.in
 +++ b/Makefile.in
-@@ -613,7 +613,7 @@ CLEANFILES = $(VC6_LIBDSP) $(VC6_SRCDSP)
+@@ -612,7 +612,7 @@ CLEANFILES = $(VC6_LIBDSP) $(VC6_SRCDSP)
  
  bin_SCRIPTS = curl-config
  SUBDIRS = lib src include
diff --git a/lede/package/network/utils/curl/patches/310-mbedtls-disable-runtime-version-check.patch b/lede/package/network/utils/curl/patches/310-mbedtls-disable-runtime-version-check.patch
index 22b838daa0..89e057eb91 100644
--- a/lede/package/network/utils/curl/patches/310-mbedtls-disable-runtime-version-check.patch
+++ b/lede/package/network/utils/curl/patches/310-mbedtls-disable-runtime-version-check.patch
@@ -1,6 +1,6 @@
 --- a/lib/vtls/mbedtls.c
 +++ b/lib/vtls/mbedtls.c
-@@ -724,7 +724,7 @@ void Curl_mbedtls_session_free(void *ptr
+@@ -729,7 +729,7 @@ void Curl_mbedtls_session_free(void *ptr
  
  size_t Curl_mbedtls_version(char *buffer, size_t size)
  {
diff --git a/lede/package/network/utils/curl/patches/100-CVE-2017-2629.patch b/lede/package/network/utils/curl/patches/100-CVE-2017-2629.patch
deleted file mode 100644
index f2cd869c5b..0000000000
--- a/lede/package/network/utils/curl/patches/100-CVE-2017-2629.patch
+++ /dev/null
@@ -1,33 +0,0 @@
-From a00a42b4abe8363a46071bb3b43b1b7138f5259b Mon Sep 17 00:00:00 2001
-From: Daniel Stenberg <daniel@haxx.se>
-Date: Sun, 22 Jan 2017 18:11:55 +0100
-Subject: [PATCH] TLS: make SSL_VERIFYSTATUS work again
-
-The CURLOPT_SSL_VERIFYSTATUS option was not properly handled by libcurl
-and thus even if the status couldn't be verified, the connection would
-be allowed and the user would not be told about the failed verification.
-
-Regression since cb4e2be7c6d42ca
-
-CVE-2017-2629
-Bug: https://curl.haxx.se/docs/adv_20170222.html
-
-Reported-by: Marcus Hoffmann
----
- lib/url.c | 3 +++
- 1 file changed, 3 insertions(+)
-
---- a/lib/url.c
-+++ b/lib/url.c
-@@ -4141,8 +4141,11 @@ static struct connectdata *allocate_conn
-   conn->bits.ftp_use_epsv = data->set.ftp_use_epsv;
-   conn->bits.ftp_use_eprt = data->set.ftp_use_eprt;
- 
-+  conn->ssl_config.verifystatus = data->set.ssl.primary.verifystatus;
-   conn->ssl_config.verifypeer = data->set.ssl.primary.verifypeer;
-   conn->ssl_config.verifyhost = data->set.ssl.primary.verifyhost;
-+  conn->proxy_ssl_config.verifystatus =
-+    data->set.proxy_ssl.primary.verifystatus;
-   conn->proxy_ssl_config.verifypeer = data->set.proxy_ssl.primary.verifypeer;
-   conn->proxy_ssl_config.verifyhost = data->set.proxy_ssl.primary.verifyhost;
- 
